src/wallet/wallettool.cpp

```diff
@@ -103,6 +104,27 @@ static void WalletShowInfo(CWallet* wallet_instance)
     tfm::format(std::cout, "Address Book: %zu\n", wallet_instance->m_address_book.size());
 }
 
+static bool SalvageWallet(const fs::path& path)
+{
+    // Create a Database handle to allow for the db to be initialized before recovery
+    std::unique_ptr<WalletDatabase> database = WalletDatabase::Create(path);
+
+    // Initialize the environment before recovery
+    bilingual_str error_string;
+    try {
+        WalletBatch::VerifyEnvironment(path, error_string);
+    } catch (const fs::filesystem_error& e) {
+        error_string = Untranslated(strprintf("Error loading wallet. %s", fsbridge::get_filesystem_error_message(e)));
+    }
+    if (!error_string.original.empty()) {
+        tfm::format(std::cerr, "Failed to open wallet for salvage :%s\n", error_string.original);
```

Comment:
```suggestion
        tfm::format(std::cerr, "Failed to open wallet for salvage: %s\n", error_string.original);
```

Reply:
Will fix if I have to push another change.